src/kernel.cpp

```diff
@@ -513,3 +489,10 @@ bool CheckStakeModifierCheckpoints(int nHeight, unsigned int nStakeModifierCheck
     }
     return true;
 }
+
+// Timestamp mask (timeprotocol V2: 15 - last four bits)
+int64_t GetMaskedTime()
+{
+    return (GetAdjustedTime() & ~int64_t(Params().StakeTimestampMask()));
```

Comment:
If we want to keep 60 seconds block time, this could be `GetAdjustedTime() / int64_t(Params().StakeTimestampMask()) * int64_t(Params().StakeTimestampMask())` with Params().StakeTimestampMask() being 15.